exec/java-exec/src/test/java/org/apache/drill/exec/impersonation/TestImpersonationQueries.java

```diff
@@ -258,8 +286,35 @@ public void testMultiLevelImpersonationJoinOneSideExceedsMaxUserHops() throws Ex
 
     assertNotNull("UserRemoteException is expected", ex);
     assertThat(ex.getMessage(),
-        containsString("Cannot issue token for view expansion as issuing the token exceeds the maximum allowed number " +
-            "of user hops (3) in chained impersonation"));
+      containsString("Cannot issue token for view expansion as issuing the token exceeds the maximum allowed number " +
+        "of user hops (3) in chained impersonation"));
+  }
+
+  @Test
+  public void sequenceFileChainedImpersonationWithView() throws Exception {
+    // create a view named "simple_seq_view" on "simple.seq". View is owned by user0:group0 and has permissions 750
+    createView(org1Users[0], org1Groups[0], "simple_seq_view",
+      String.format("SELECT convert_from(t.binary_key, 'UTF8') as k FROM %s.`%s` t", MINIDFS_STORAGE_PLUGIN_NAME,
+        new Path(getUserHome(org1Users[0]), "simple.seq")));
+    try {
+      updateClient(org1Users[1]);
+      test("SELECT k FROM %s.%s.%s", MINIDFS_STORAGE_PLUGIN_NAME, "tmp", "simple_seq_view");
+    } catch (UserRemoteException e) {
+      assertNull("This test should pass.", e);
+    }
+  }
+
+  @Test
+  public void avroChainedImpersonationWithView() throws Exception {
```

Comment:
I am curious.. Did this test fail without the impersonation changes in AvroRecordReader?